### ISSUE TITLE
Error: statefulsets.apps "sentry-clickhouse" is forbidden: 

### ISSUE THREAD

**Issue by cdesch** [sentry-kubernetes/charts#492]
Has anyone encountered this error before when deploying sentry? I was deploying manually with:

     helm install sentry sentry -n sentry -f sentry/values.yaml

Error Message:

     Error: statefulsets.apps "sentry-clickhouse" is forbidden: non-admin user "3432-2342-23423432-234234"
     [service account "sentry:default"]. The configured privileged attributes access for non-admin users
     ("[]")("[]") and for service accounts ("[]")("[]") lack required permissions to use attributes [privileged] 
     for resource sentry-clickhouse

**Comment by sagor999** [comment]
clickhouse is deployed with:
```
securityContext:    
  privileged: true  
  runAsUser: 0      
```
which seems like your cluster is not allowing. I would check with whoever manages your cluster to see if they can allow clickhouse to get deployed with privileged permissions.

**Comment by cdesch** [comment]
Why does Clickhouse need this priviedge? I can try working with the cluster admins on this, but if it is not something that is doable, can clickhouse be disabled?

**Comment by sagor999** [comment]
That's a good question. 
It seems like there is no "official" helm chart for clickhouse. Sentry uses helm chart that seems to be cloned from this repo: https://github.com/liwenhe1993/charts/tree/master/clickhouse
which has not been updated for two years.

I am not sure if there is an actual reason for clickhouse to run with privileged access, but maybe @Mokto knows.

**Comment by Mokto** [comment]
To be honest, no idea!

**Comment by cdesch** [comment]
Thanks @Mokto @sagor999 !

I'm not familiar with clickhouse, so I could be wrong here, but I'm still not sure why clickhouse would need the elevated privileges. I tried forking the [clickhouse helmchart](https://github.com/liwenhe1993/charts) and commenting out: 

```
securityContext:    
  privileged: true  
  runAsUser: 0      
```

and things appear to work fine on my local machine, but that doesn't take into context if `sentry-kubernetes` has a requirement for clickhouse to have the elevated permissions.

While I couldn't find another helm chart for clickhouse, I did find this helm chart for [PostHog](https://github.com/PostHog/charts-clickhouse) that has a [clickhouse instance running](https://github.com/PostHog/charts-clickhouse/tree/main/charts/posthog/templates). In this [deployment](https://github.com/PostHog/charts-clickhouse/blob/9b74bb2a90a9c2dbde8c4cfef18f866c9329856f/charts/posthog/templates/clickhouse_instance.yaml#L82), clickhouse runs as [` runAsUser: 101` ](https://www.botkube.io/configuration/helm-options/) which is a non-privledged account. I believe the PostHog helm chart is using this [clickhouse Operator (CRD)](https://github.com/Altinity/clickhouse-operator) 

Lastly, the version of clickhouse from the current helm chart is `19.14`, which is more than a few years behind the current release of clickhouse `21.9.4.35`. I'm not in favor of upgrading for the sake of upgrading, but it would be nice to have a path for upgrading in the event of vulnerabilities. 

Would it be worth considering switching out the clickhouse helm chart for the click helm chart? I'm not sure how heavy of a lift this would be, but I'm willing to help.

**Comment by sagor999** [comment]
I think switching helm chart might be very disruptive if new chart is using different naming convention, as in that case you might not be able to re-use PVCs from old chart (or might have to specify them manually in new chart if they allow it). 
But what we could do it upgrade clickhouse image that is used by the helm chart.
I looked at sentry onpremise docker compose and they are using slightly newer version of clickhouse:
`image: "yandex/clickhouse-server:20.3.9.70"`

Would just need to double check if there is any migration that needs to be done after upgrading, or if all is needed is to bump container version. 

Edit: and we can update our clickhouse helm chart to not use privileged account. I think that would be the first step. And then upgrade to match version that is used by sentry onpremise.

**Comment by cdesch** [comment]
I think that would be a great first step. 

I'm still learning helm, but I can give it a shot. If I make the changes to `templates/*` and `values.yaml` in the [`clickhouse`](https://github.com/sentry-kubernetes/charts/tree/develop/clickhouse) directory, will the [`clickhouse-3.0.0.tgz`](https://github.com/sentry-kubernetes/charts/tree/develop/sentry/charts) need to updated too to do a manual: `helm install sentry sentry -n sentry -f sentry/values.yaml`?

**Comment by sagor999** [comment]
https://github.com/sentry-kubernetes/charts/blob/develop/clickhouse/Chart.yaml
You need to bump the version of clickhouse chart. Since it is a minor change, I would bump it to 3.1.0.
And then you run `helm dependency update` in sentry chart so that it will generate a new tgz file for clickhouse.

**Comment by cdesch** [comment]
Thanks @sagor999 

I get this error: 

```
cjsMBP15:sentry cj1$ helm dependency update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "vulcanlink" chart repository
...Successfully got an update from the "sentry" chart repository
...Successfully got an update from the "bitnami" chart repository
Update Complete. ⎈Happy Helming!⎈
Error: can't get a valid version for repositories clickhouse. Try changing the version constraint in Chart.yaml
```

I think the update needs to be committed to the repo before it will allow me to do a `helm dependency update`

Here are my changes: https://github.com/cdesch/charts/commit/1b1198aa91a6c090161eb9eb8d988ffcebd9a3fa

Should I make a PR?

**Comment by sagor999** [comment]
Do one change at a time. 
So submit a PR for removing privileged from clickhouse chart first (bump clickhouse chart to 3.1.0). Once that gets approved, another PR can bump the chart version of clickhouse that is used in sentry chart, and hopefully that can be tested first for a bit. 
Once that is merged and all is well, another PR can bump the version of clickhouse container image. :)

**Comment by cdesch** [comment]
Thanks. First PR is  [493](https://github.com/sentry-kubernetes/charts/pull/493)

I bumped the version of the container image in this PR. My mistake for not reading. I can redo it if you would like.

**Comment by cdesch** [comment]
I reverted the clickhouse container image version in the PR

**Comment by cdesch** [comment]
@sagor999 Should I make the second PR in the meantime?

**Comment by sagor999** [comment]
I would wait until that one got approved and merged first.

**Comment by cdesch** [comment]
Thanks you. 

@Mokto Please let me know if you have any feedback for the PR 😸

**Comment by z0rc** [comment]
Are you still planning to update clickhouse version? I have a hunch, that version bump should fix issues seen in #487.

**Comment by cdesch** [comment]
@z0rc I'm going to try bumping it to `imageVersion: "20.3.9.70"` after I get this in. Then, possibly moving clickhouse chart to `3.2.0` and updating everything. I'm testing the new config of clickhouse today.

**Comment by z0rc** [comment]
I just noticed, even as clickhouse chart defaults to `19.14` image. Sentry chart overrides this value here https://github.com/sentry-kubernetes/charts/blob/12.0.0/sentry/values.yaml#L642. So I'm not sure how bumping clickhouse chart value might help in this specific case.

**Comment by cdesch** [comment]
I actually saw that too. Bumping the version in clickhouse won't do anything since it is overridden in the `values.yaml`. If anything we could bump the version in clickhouse to `20.8.9.6`, which it is currently at in  `values.yaml`, as part of some housekeeping.

**Comment by Mokto** [comment]
This issue is stale because it has been open for 30 days with no activity.

**Comment by ba2512005** [comment]
Hi Team, did this get resolved in 20.8.9.6 to not use privileged container? Or is there a version that works without having privileged access?

**Comment by Mokto** [comment]
This issue is stale because it has been open for 30 days with no activity.

**Comment by Mokto** [comment]
This issue was closed because it has been inactive for 14 days since being marked as stale.
